### PR TITLE
[AKS] `az aks update` : Support edit nodepool label after creation 

### PR DESCRIPTION
**Pull request by Xinyue-Wang**
**Description**

Add support to allow edit label after creation.
ex: 
az aks nodepool update --labels foo=bar
az aks update --nodepool-labels foo=bar

**Testing Guide**

**History Notes**

[AKS] `az aks update`: Support edit nodepool label after creation 
[AKS] `az aks nodepool update`: Support edit nodepool label after creation 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by wangzelin007 — APPROVED** [review]
Please add recording files, others LGTM.

**Comment by andyliuliming** [comment]
+ @fumingzhang to review.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -5884,4 +5973,47 @@ def test_aks_enable_addon_with_azurekeyvaultsecretsprovider(self, resource_group
         cmd = 'aks delete --resource-group={resource_group} --name={name} --yes --no-wait'
         self.cmd(cmd, checks=[
             self.is_empty(),
+        ])
+
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus2')
+    @AKSCustomRoleBasedServicePrincipalPreparer()
+    def test_aks_update_labels(self, resource_group, resource_group_location, sp_name, sp_password):
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:5981]: This test case will fail in live mode ([log](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=48698070&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=dc59ccd1-231f-538b-777f-33a592c7ca57&l=414)).

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -2728,6 +2730,9 @@ def aks_update(cmd, client, resource_group_name, name,
                 '--rotation-poll-interval can only be specified when azure-keyvault-secrets-provider is enabled')
         azure_keyvault_secrets_provider_addon_profile.config[CONST_ROTATION_POLL_INTERVAL] = rotation_poll_interval
 
+    if nodepool_labels is not None:
+        instance.node_labels = nodepool_labels
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2734]: The reason of the failed test case should be the `instance` here is the `ManagedCluster` object but not an `AgentPool` object. You may change this to something like `instance.agent_pool_profiles[0].node_labels = nodepool_labels`. May consider some other checks.

**Review by FumingZhang — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -2728,6 +2730,10 @@ def aks_update(cmd, client, resource_group_name, name,
                 '--rotation-poll-interval can only be specified when azure-keyvault-secrets-provider is enabled')
         azure_keyvault_secrets_provider_addon_profile.config[CONST_ROTATION_POLL_INTERVAL] = rotation_poll_interval
 
+    if nodepool_labels is not None:
+        for agent_profile in instance.agent_pool_profiles:
+            agent_profile.node_labels = nodepool_labels
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2735]: This would update all the pools, is this expected? Do we need to remind users?

Comment (Xinyue-Wang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2735]: Yes. If customer use `aks update --nodepool-labels` to update labels, it means labels on all node pools will be updated to the same value. if they wish to update different label on different node pool separately, they should use `aks nodepool update --labels`. We can make this clear in our doc when add these new commands.

**Review by FumingZhang — APPROVED** [review]

**Comment by Xinyue-Wang** [comment]
@andyliuliming @zhoxing-ms @jsntcy Could you help review the pull request? My pr still need an owner approval to merge

**Comment by FumingZhang** [comment]
Hi @Xinyue-Wang, I notice that we do not have these changes included in aks-preview. To ensure that users can also update the node label after installing aks-preview (which would totally override azure-cli), could you please open another PR to apply these changes to aks-preview?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**